kotlin-checks-test-sources/src/main/kotlin/checks/MainSafeCoroutinesCheckSample.kt

```diff
@@ -0,0 +1,117 @@
+import java.io.IOException
+import kotlin.jvm.Throws
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+
+class MainSafeCoroutinesCheckSample {
+
+    suspend fun throwsIO3() {
+        GlobalScope.launch {
+            f() // Noncompliant {{Use "Dispatchers.IO" to run this potentially blocking operation}}
+        }
+        delay(500L)
+    }
+
+    suspend fun throwsIO4() {
+        withContext(Dispatchers.Default) {
+            f() // Noncompliant {{Use "Dispatchers.IO" to run this potentially blocking operation}}
+        }
+        delay(500L)
+    }
+    suspend fun throwsIO() {
+        f() // Noncompliant {{Use "Dispatchers.IO" to run this potentially blocking operation}}
+        delay(500L)
+    }
+
+    suspend fun throwsIO5() {
+        withContext(Dispatchers.Main) {
+            f() // Noncompliant {{Use "Dispatchers.IO" to run this potentially blocking operation}}
+        }
+        delay(500L)
+    }
+
+    suspend fun throwsIO6(context: CoroutineDispatcher) {
+        withContext(context) {
+            f() // Compliant
+        }
+        delay(500L)
+    }
+
+    suspend fun throwsIO7(obj: Any?) {
+        withContext(Dispatchers.Main) {
+            obj?.let {
+                f() // Noncompliant
+            }
+        }
+        delay(500L)
+    }
+
+    suspend fun throwsIO8(obj1: Any?, obj2: Any?, obj3: Any?) {
+        obj1?.let {
+            obj2?.let {
+                obj3?.let {
+                    f() // Noncompliant
+                }
+            }
+        }
+        delay(500L)
+    }
+
+    // False-negative, can't get throwing section from binaries
+    suspend fun javaThrowsIO2() {
+        MyFile().createFile()
+        delay(500L)
+    }
+
+    suspend fun suspendThreadSleep() {
+        Thread.sleep(500L) // Noncompliant {{Replace this "Thread.sleep()" call with "delay()"}}
+    }
+
+    fun threadSleep() {
+        Thread.sleep(500L) // Compliant, not in a suspending function
+    }
+
+    fun inSuspendingLambda() {
+        runBlocking {
+            Thread.sleep(500L) // Noncompliant {{Replace this "Thread.sleep()" call with "delay()"}}
+        }
+    }
+
+    fun withGlobalScope() {
+        GlobalScope.launch {
+            Thread.sleep(500L) // Noncompliant {{Replace this "Thread.sleep()" call with "delay()"}}
+        }
+    }
+
+    fun inSuspendingLambdaDelay() {
+        runBlocking {
+            delay(500L)
+        }
+    }
+    // False-negative, can't get throwing section from binaries
+    suspend fun javaThrowsIO() {
+        File("").getCanonicalPath()
+        delay(500L)
+    }
+
+    suspend fun compliant() {
+        withContext(Dispatchers.IO) {
+            f()
+        }
+        delay(500L)
+    }
+}
+
+@Throws(IOException::class)
+fun f() {
+}
```

Comment:
How about a compliant `Thread.sleep` in a non-suspending fun, to test we don't have FPs there?
Same for a call to `f()`?

Reply:
Is it what you're talking about? [L79-L81](https://github.com/SonarSource/sonar-kotlin/pull/28/files/664bb3413d9500ff73c4f7b37f274870518f541b#diff-76cc128772b9466def676256139e9ebd35c885f751dc6230da1ac46003ad6a43R79-R81)

Reply:
My bad, yes!